### PR TITLE
Added support for QueryManual and UpdateManual

### PR DESCRIPTION
**Pull request by dansteren**
This gets the ic_api example working. `QueryManual` and `UpdateManual` canister methods are now supported... as long as they always reject. Calling `ic.reply` is not yet implemented, that will come next.

Addresses part of #655

### CODE REVIEW

File: src/compiler/typescript_to_rust_redesign/azle_generate/src/generators/canister_methods/method_body.rs [hunk 1]
```diff
@@ -43,3 +47,54 @@ pub fn generate_call_to_js_function(fn_decl: &FnDecl) -> proc_macro2::TokenStrea
...
+    }
+
+    quote! {
+        _azle_final_return_value.azle_try_from_js_value(&mut _azle_boa_context).unwrap()
+    }
+}
+
+/// Returns true if the return type is `empty`, `never`, `null`, `void`, or `undefined`.
+/// Otherwise returns false.
+fn type_is_emptyish(ts_type_option: Option<&TsType>) -> bool {
```
Comment (lastmjs) [inline src/compiler/typescript_to_rust_redesign/azle_generate/src/generators/canister_methods/method_body.rs:80]: Get rid of never and undefined, make a comment TODO to consider that functionality

File: src/compiler/typescript_to_rust_redesign/azle_generate/src/generators/ic_object/functions/reply.rs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+fn generate_match_arm(fn_decl: &FnDecl) -> TokenStream {
+    let fn_name = utils::fn_decls::get_fn_decl_function_name(fn_decl);
+    let quoted_fn_name: TokenStream = format!("\"{}\"", fn_name).parse().unwrap();
+    let return_type_ast = utils::fn_decls::get_canister_method_return_type(fn_decl);
+    let rust_return_type = match return_type_ast {
+        Some(ts_type) => canister_methods::ts_type_to_rust_type(ts_type, None).get_type_ident(),
+        None => quote! {()},
+    };
+    quote!(
+        #quoted_fn_name => {
```
Comment (lastmjs) [inline src/compiler/typescript_to_rust_redesign/azle_generate/src/generators/ic_object/functions/reply.rs:44]: Just interpolate fn_name directly

**Review by lastmjs — CHANGES_REQUESTED** [review]

File: src/compiler/typescript_to_rust_redesign/azle_generate/src/generators/canister_methods/method_body.rs (hunk 1 above)
Comment (dansteren) [reply src/compiler/typescript_to_rust_redesign/azle_generate/src/generators/canister_methods/method_body.rs:80]: :+1: Done!

File: src/compiler/typescript_to_rust_redesign/azle_generate/src/generators/ic_object/functions/reply.rs (hunk 2 above)
Comment (dansteren) [reply src/compiler/typescript_to_rust_redesign/azle_generate/src/generators/ic_object/functions/reply.rs]: Thanks for teaching me this. Done!

**Review by lastmjs — APPROVED** [review]

**Pull request merged**